src/validate.ts

```diff
@@ -1,113 +1,43 @@
 'use-strict';
+import * as Ajv from 'ajv';
 
-/**
- * Generates random request ID 
- * @param min Minimum number
- * @param max Maximum number 
- * @returns Random number between the range of the min to max number specified
- */
-function generateRequestID(min: number, max: number){
-     return Math.floor(Math.random() * (max-min) + min).toString()
-}
-
-/**
- * Generates SYNC request 
- * @returns Specified format for SYNC intent request. 
- */
-export function generateSyncRequest(){
-    const requestId = generateRequestID(100,999);
-    return {
-        requestId,
-        inputs: [{
-          intent: "action.devices.SYNC"
-        }]
-    }
-}
-
-interface Device {
-  id: string;
-  customData?: {[key: string]: any};
-}
+const ajv = new Ajv({
+  allErrors: true,
+});
 
-interface QueryInput {
-  intent: string; 
-  payload: {
-    devices: Device[]; 
-  };
-}
-
-interface QueryRequest {
-  requestId: string;
-  inputs: QueryInput[];
-}
-
-/**
- * Generates QUERY request 
- * @param devices Array of type Device containing device IDs and respective custom data
- * @returns Specified format for QUERY intent request. 
- */
-export function generateQueryRequest(devices: Device[]) : QueryRequest{
-    const requestId = generateRequestID(100,999);
-    return {
-        requestId,
-        inputs: [{
-          intent: "action.devices.QUERY",
-          payload: {
-            devices
-            }
-        }]
-    }
-}
-
-
-interface Command{
-    name: string,
-    params?: {[key: string]: any}
-}
-
-interface ExecuteInput{
-    intent: string;
-    payload: {
-        devices: Device[];
-        execution: Command[];
-    };
-}
-
-interface ExecuteRequest{
-    requestId: string;
-    inputs: ExecuteInput[];
-}
+const syncResponseSchema = require('../intents/sync.response.schema.json');
```

Comment:
Typically the [style for constants](https://google.github.io/styleguide/jsguide.html#naming-constant-names) is UPPERCASE_UNDERSCORES.